ng2-components/ng2-alfresco-datatable/src/components/datatable/datatable.component.ts

```diff
@@ -88,16 +111,22 @@ export class DataTableComponent implements AfterContentInit {
 
         if (!this.data) {
             this.data = new ObjectDataTableAdapter([], schema);
-        } else if (schema && schema.length > 0) {
-            this.data.setColumns(schema);
+        } else {
+            this.setHtmlColumnConfigurationOnObjectAdapter(schema);
         }
 
-         // workaround for MDL issues with dynamic components
+        // workaround for MDL issues with dynamic components
         if (componentHandler) {
             componentHandler.upgradeAllRegistered();
         }
     }
 
+    setHtmlColumnConfigurationOnObjectAdapter(schema: DataColumn[]) {
```

Comment:
Can we at least make it `private`? As for me this is worse than leaving a comment, if component structure is changed we will have to rename this public method, refactor unit tests and change documentation.